### ISSUE TITLE
extra_data doesn't always arrive to logic adapter

### ISSUE THREAD

**Issue by thedp** [gunthercox/ChatterBot#593]
I'm using the django as in/out.
I've noticed that extra_data doesn't always arrive to my custom logic_adapter. But when I delete the sqlite3 db, it starts working again, and then stops again.

Is this by design?

Thank you.

**Comment by gunthercox** [comment]
Are you saving a statement in your custom logic adapter? The `extra_data` attribute is empty by default so if a new statement gets created and that statement has the same text as an existing one, then when it gets saved the empty value for `extra_data` would replace the existing value.

**Comment by thedp** [comment]
I'm not intentionally saving.
I'm talking only about the statement from the client. You say after recieving the exact same statement, the extra_data doesn't get passed to the logic adapter?

Is the only way to always get the extra_data from the client is by setting chatterbot to read_only=True ?

**Comment by gunthercox** [comment]
Setting `read_only=True` just controls if the chat bot can "learn", if it is set to true then none of of the input the chat bot gets will be saved to the database.

It sounds like this might be a bug. I will test it out and make changes to make sure it gets fixed.

**Comment by gunthercox** [comment]
@thedp It's been a while, but I just introduced a change that will be available in the next version of ChatterBot which will inadvertently correct this issue. The `extra_data` attribute on statements has been removed in favor of a `tags` attribute. Additional testing has been done to ensure that tags perform as expected.

I'm going to close this ticket off for now, but feel free to create a new one if you have any questions.
